### PR TITLE
Fixed docstrings in exceptions.py, ran pep8 and pep257.

### PR DESCRIPTION
**Pull request by terryjbates**
Working on #742.

### CODE REVIEW

File: cookiecutter/exceptions.py [hunk 1]
```diff
@@ -1,104 +1,84 @@
...
-    this class.
+    """Base exception class.
+
+    All Cookiecutter-specific exceptions should subclass this class.
     """
 
 
 class NonTemplatedInputDirException(CookiecutterException):
-    """
-    Raised when a project's input dir is not templated.
```
Comment (michaeljoseph) [inline cookiecutter/exceptions.py]: I have a preference for the summary line of [multi-line docstrings](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings) to be on the next line.
>  The summary line may be on the same line as the opening quotes _or on the next line_. 

Thoughts @audreyr @pydanny @hackebrot

File: cookiecutter/exceptions.py [hunk 2]
```diff
@@ -1,104 +1,84 @@
...
     The name of the input directory should always contain a string that is
     rendered to something else, so that input_dir != output_dir.
     """
 
 
 class UnknownTemplateDirException(CookiecutterException):
-    """
-    Raised when Cookiecutter cannot determine which directory is the project
-    template, e.g. more than one dir appears to be a template dir.
+    """Raised when directory that is project template cannot determined.
```
Comment (michaeljoseph) [inline cookiecutter/exceptions.py]: Raised when the project template directory cannot be determined.

File: cookiecutter/exceptions.py [hunk 3]
```diff
@@ -1,104 +1,84 @@
...
+
+    More than one dir appears to be a template dir.
     """
 
 
 class MissingProjectDir(CookiecutterException):
-    """
-    Raised during cleanup when remove_repo() can't find a generated project
-    directory inside of a repo.
+    """Raised during cleanup stages when project directory appears missing.
```
Comment (michaeljoseph) [inline cookiecutter/exceptions.py]: .. when **the** project directory **is** missing.

File: cookiecutter/exceptions.py [hunk 4]
```diff
@@ -1,104 +1,84 @@
...
+
+    When remove_repo() cannot locate generated project directory inside repo.
```
Comment (michaeljoseph) [inline cookiecutter/exceptions.py]: ...cannot locate **the** generated project directory.

File: cookiecutter/exceptions.py [hunk 5]
```diff
@@ -1,104 +1,84 @@
...
+    """Raised when get_config() is passed a invalid config file path."""
 
 
 class InvalidConfiguration(CookiecutterException):
-    """
-    Raised if the global configuration file is not valid YAML or is
-    badly contructed.
+    """Raised if the global configuration file is not valid.
+
+    Invalid YAML or is configuration file is badly constructed.
```
Comment (michaeljoseph) [inline cookiecutter/exceptions.py]: Invalid YAML or **the configuration** file is badly constructed.

File: cookiecutter/exceptions.py (hunk 1 above)
Comment (hackebrot) [inline cookiecutter/exceptions.py]: I'm fine with both styles. I personally use:

```python
"""Decorator that produces DEBUG level log messages before and after
calling a parser method.

If a callback raises an IgnoredMatchException the log will show 'IGNORED'
instead to indicate that the parser will not create any objects from
the matched string.
"""
```

and 

```python
"""Helper to log an escaped version of the given message to DEBUG"""
```

**Comment by terryjbates** [comment]
Thanks to all for the feedback! @michaeljoseph I'm making mods on all the changes with the exception of the first. I _think_ I savaged the summary line as such in the first change as such, since I could not craft language to make this fall under 80 chars for pep8.  Also, pep257 complains if the summary line does not end in an explicit "."  I'm not personally wedded to either style, but this may be why I angled for a summary line, with fuller detail on next line. Thoughts? I will go ahead and push the other changes to this branch momentarily.

**Comment by hackebrot** [comment]
Hey @terryjbates! Your last commit's email does not seem to be recognized by GitHub.

**Comment by terryjbates** [comment]
Think I know why. Did the commit from an odd computer that lacked my proper git config info. Sorry about that. Will see if I can set this locally to proper email, amend, and push.

File: cookiecutter/exceptions.py [hunk 6]
```diff
@@ -1,104 +1,84 @@
...
+    """Raised when template uses an undefined variable in the context."""
+
     def __init__(self, message, error, context):
+        """Initialize instance of UndefinedVariableInTemplate class."""
         self.message = message
         self.error = error
         self.context = context
 
     def __str__(self):
+        """Create informal string representation of object."""
```
Comment (michaeljoseph) [inline cookiecutter/exceptions.py]: I think we should be suppressing these magic method docstrings.

File: cookiecutter/exceptions.py [hunk 7]
```diff
@@ -1,104 +1,84 @@
...
 
 
 class UndefinedVariableInTemplate(CookiecutterException):
-    """Raised when a template uses a variable which is not defined in the
-    context.
-    """
+    """Raised when template uses an undefined variable in the context."""
+
     def __init__(self, message, error, context):
+        """Initialize instance of UndefinedVariableInTemplate class."""
```
Comment (michaeljoseph) [inline cookiecutter/exceptions.py]: @terryjbates could you please add parameter documentation here?

**Comment by michaeljoseph** [comment]
@terryjbates 🔔

**Comment by terryjbates** [comment]
@michaeljoseph Sure thing, I can look into adding param doc.

**Pull request closed without merging**